### PR TITLE
DOC: clarify installation

### LINKED ISSUE #257: suggestions for clarifying hnn_core installation instructions
1) The link to the anaconda installation is not working and goes to an error page 
2) explain that if this is the first time installing hnn_core that "--update" should be omitted in the pip install command 
3) explain that if you have an older version of NEURON install, you will first have to remove it. And, then the most up to date version of NEURON should be installed via pip and not the automatic NEURON installation package

### PR DESCRIPTION
**Pull request by jasmainak**
cc @stephanie-r-jones let me know if something else needs to be clarified

closes #257

### CODE REVIEW

**Comment by jasmainak** [comment]
I decided to leave the `--upgrade` in there because it should work even if it's a first time installation

File: README.rst [hunk 1]
```diff
@@ -41,14 +41,21 @@ Optional dependencies
 Installation
 ============
 
-We recommend the `Anaconda Python distribution <https://www.continuum.io/downloads>`_. To install ``hnn-core``, you first need to install its dependencies::
+We recommend the `Anaconda Python distribution <https://www.anaconda.com/products/individual>`_.
```
Comment (cjayb) [inline README.rst:44]: Should we mention a recommended Python minor version? I guess that largely depends on what NEURON is built against.

File: README.rst [hunk 2]
```diff
@@ -41,14 +41,21 @@ Optional dependencies
...
+Additionally, you would need Neuron which is available here:
+`https://neuron.yale.edu/neuron/ <https://neuron.yale.edu/neuron/>`_.
+We recommend that you install it via pip::
 
 	$ pip install NEURON
 
+Note that if you choose to install Neuron using the traditional installer, it may be hard
+to manage it with virtual environments in Anaconda. For this reason, we recommend the
+pip installer.
+
```
Comment (cjayb) [inline README.rst:58]:
```suggestion
Note installing Neuron using the traditional installer may make it hard to manage
within virtual environments such as Anaconda. For this reason, we recommend the
pip installer:

	$ pip install NEURON

```

**Review by cjayb — APPROVED** [review]
Minor stylistic suggestion, other than that GTG.

File: README.rst (hunk 1 above)
Comment (jasmainak) [reply README.rst:44]: I think Neuron doesn't work with 3.9 as far as I know ... I hope Anaconda doesn't distribute bleeding edge Python

Comment (cjayb) [reply README.rst:44]: It might... We could recommend creating an env with python<=3.8? Gets a bit messy though.

Comment (jasmainak) [reply README.rst:44]: yeah let's leave it like this for now and see how many users stumble upon that issue

**Comment by jasmainak** [comment]
Merged as recommended by @cjayb in #260

**Pull request merged**